components/org.wso2.identity.event.websubhub.publisher/src/main/java/org/wso2/identity/event/websubhub/publisher/util/WebSubHubAdapterUtil.java

```diff
@@ -223,4 +237,149 @@ public static void logPublishingEvent(String url, EventContext eventContext, Str
+                                              String callbackUrl)
+            throws WebSubAdapterServerException {
+
+        try {
+            URIBuilder uriBuilder = new URIBuilder(webSubHubBaseUrl);
+            uriBuilder.addParameter(HUB_MODE, operation);
+            uriBuilder.addParameter(HUB_TOPIC, topic);
+            uriBuilder.addParameter(HUB_CALLBACK, callbackUrl);
+            return uriBuilder.build().toString();
+        } catch (URISyntaxException e) {
+            log.error("Error building subscription URL", e);
+            throw handleServerException(ERROR_INVALID_WEB_SUB_HUB_BASE_URL, e);
+        }
+    }
+
+    /**
+     * Construct the hub topic by combining tenant domain and topic suffix.
+     *
+     * @param topicSuffix  Topic suffix.
+     * @param tenantDomain Tenant domain.
+     * @return Hub topic.
+     */
+    public static String constructHubTopic(String topicSuffix, String tenantDomain) {
+
+        return tenantDomain + WebSubHubAdapterConstants.Http.TOPIC_SEPARATOR + topicSuffix;
+    }
+
+    /**
+     * Get the WebSub Hub base URL from configuration.
+     *
+     * @return WebSub Hub base URL.
+     * @throws WebSubAdapterException If the WebSub Hub base URL is not configured.
+     */
+    public static String getWebSubBaseURL() throws WebSubAdapterException {
+
+        String webSubHubBaseUrl =
+                WebSubHubAdapterDataHolder.getInstance().getAdapterConfiguration().getWebSubHubBaseUrl();
+
+        if (StringUtils.isEmpty(webSubHubBaseUrl)) {
+            throw handleClientException(WebSubHubAdapterConstants.ErrorMessages.WEB_SUB_BASE_URL_NOT_CONFIGURED);
+        }
+        return webSubHubBaseUrl;
+    }
+
+    /**
+     * Handle successful WebSub Hub operations.
+     *
+     * @param entity    HTTP entity.
+     * @param topic     Topic.
+     * @param operation Operation.
+     * @throws WebSubAdapterException If an error occurs while handling the response.
+     * @throws IOException            If an error occurs while reading the response.
+     */
+    public static void handleSuccessfulOperation(HttpEntity entity, String topic, String operation)
+            throws WebSubAdapterException, IOException {
+
+        if (entity != null) {
+            String responseString = EntityUtils.toString(entity, StandardCharsets.UTF_8);
+            if (RESPONSE_FOR_SUCCESSFUL_OPERATION.equals(responseString)) {
+                log.debug("Success WebSub Hub operation: " + operation + ", topic: " + topic);
+            } else {
+                throw handleServerException(ERROR_INVALID_RESPONSE_FROM_WEBSUB_HUB, null,
+                        topic, operation, responseString);
+            }
+        } else {
+            String message = String.format(ERROR_EMPTY_RESPONSE_FROM_WEBSUB_HUB.getDescription(), topic, operation);
+            throw new WebSubAdapterServerException(message, ERROR_EMPTY_RESPONSE_FROM_WEBSUB_HUB.getCode());
+        }
+    }
```

Comment:
why we have this in a util. IMO it should go to the class that makes the call and it's a validation it seems